### DISCUSSION TITLE
Feature request - NILAN Gateway integration directly?

### DISCUSSION THREAD

**Discussion by convicte** [veista/nilan#87] [Q&A]
Hi @veista and thank you for your great work thus far.

Please excuse me if I misunderstood the way the integration is envisioned to work, though in my ignorance I was hoping that having a NILAN LAN gateway would be a positive. In other words, getting my Compact heat pump into HA would be a breeze via Modbus to TCP, since the gateway is effectively a Modbus to TCP bridge.

The more I tried to force my way through the config flow, and failing, the more I realized what you need is a generic Modbus-to-TCP bridge, and not the NILAN one. This seemingly prevents me from getting the unit into HA.

TL/DR: would you envision being able to use the NILAN bridge directly to integrate into HA?

Cheers!!

**Comment by veista** [comment]
The Nilan gateway is a propreitary cloud service. Unfortunately there is no way for me to make it work with this integration. If you get the api documentation from Nilan I may concider it, though I have no way of debugging it, so you would have to purchase me one.

**Comment by wimwes** [comment]
I like to help you with CTS700 integration. I use it now in the combination nodered with domoticz. I work on migration to home assistent.

**Comment by piwvh** [comment]
This repository integrates Nilan devices (with a Nilan Gateway) into Home Assistant:

https://github.com/superrob/genvexconnect
